### PR TITLE
planner: support create binding from existing plan

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12187
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by fzzf678 — APPROVED** [review]

File: sql-plan-management.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
 
 ### Create a binding
 
+You can create a binding for a SQL statement according to a SQL hint or a historical execution plan.
```
Comment (fzzf678) [inline sql-plan-management.md:17]:
```suggestion
You can create a binding for a SQL statement according to a SQL statement or a historical execution plan.
```

File: sql-plan-management.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ An SQL binding is the basis of SPM. The [Optimizer Hints](/optimizer-hints.md) d
...
+
+#### Create a binding according to a SQL hint
```
Comment (fzzf678) [inline sql-plan-management.md:19]:
```suggestion
#### Create a binding according to a SQL statement
```

File: sql-plan-management.md [hunk 3]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
...
+- Currently, this feature only supports binding historical execution plans in the `statements_summary` and `statements_summary_history` tables of the current TiDB node. If you get a `can't find any plans` error, you can connect to another TiDB node in the cluster and retry the binding.
+- Currently, this feature does not work for queries with subqueries, queries that access TiFlash, or queries that join 3 or more tables.
+
+The SQL statement of this binding method is as follows:
+
+```sql
+CREATE [GLOBAL | SESSION] BINDING FROM HISTORY USING PLAN DIGEST 'plan_digest';
+```
+
+This statement binds a execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. For the applicable SQL statements, priorities, scopes, and effective conditions of the created bindings, they are the same as that of [bindings created using SQL hints](#create-a-binding-according-to-a-sql-hint).
```
Comment (fzzf678) [inline sql-plan-management.md:189]:
```suggestion
This statement binds a execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. For the applicable SQL statements, priorities, scopes, and effective conditions of the created bindings, they are the same as that of [bindings created using SQL hints](#create-a-binding-according-to-a-sql-statement).
```

File: sql-statements/sql-statement-create-binding.md [hunk 4]
```diff
@@ -29,6 +30,10 @@ BindableStmt ::=
 
 ## Examples
 
+You can create a binding according to a SQL hint or a historical execution plan.
```
Comment (fzzf678) [inline sql-statements/sql-statement-create-binding.md:33]:
```suggestion
You can create a binding according to a SQL statement or a historical execution plan.
```

File: sql-statements/sql-statement-create-binding.md [hunk 5]
```diff
@@ -29,6 +30,10 @@ BindableStmt ::=
...
+
+The following example shows how to create a binding according to a SQL hint.
```
Comment (fzzf678) [inline sql-statements/sql-statement-create-binding.md:35]:
```suggestion
The following example shows how to create a binding according to a SQL statement.
```

File: sql-statements/sql-statement-drop-binding.md [hunk 6]
```diff
@@ -23,7 +24,11 @@ BindableStmt ::=
     ( SelectStmt | UpdateStmt | InsertIntoStmt | ReplaceIntoStmt | DeleteStmt )
 ```
 
-## Syntax description
+## Examples
+
+You can remove a binding according to a SQL statement or `plan_digest`.
```
Comment (fzzf678) [inline sql-statements/sql-statement-drop-binding.md:29]:
```suggestion
You can remove a binding according to a SQL statement or `sql_digest`.
```

File: sql-statements/sql-statement-drop-binding.md [hunk 7]
```diff
@@ -130,6 +135,65 @@ mysql> SHOW SESSION BINDINGS\G
 Empty set (0.00 sec)
 ```
 
+The following example shows how to remove a binding according to `plan_digest`.
```
Comment (fzzf678) [inline sql-statements/sql-statement-drop-binding.md:138]:
```suggestion
The following example shows how to remove a binding according to `sql_digest`.
```

File: sql-plan-management.md [hunk 8]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
 >
 > For `PREPARE` / `EXECUTE` statements and for queries executed with binary protocols, you need to create execution plan bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
 
-### Remove binding
+#### Create a binding according to a historical execution plan
+
+To fix a SQL statement to use a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than using a SQL hint to bind it.
```
Comment (qiancai) [inline sql-plan-management.md:171]:
```suggestion
To fix a SQL statement to use a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than using a SQL statement to bind it.
```

File: sql-plan-management.md [hunk 9]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
 >
 > For `PREPARE` / `EXECUTE` statements and for queries executed with binary protocols, you need to create execution plan bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
 
-### Remove binding
+#### Create a binding according to a historical execution plan
+
+To fix a SQL statement to use a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than using a SQL statement to bind it.
```
Comment (qiancai) [inline sql-plan-management.md:171]:
```suggestion
To fix a SQL statement to use a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than binding it according to a SQL statement.
```

File: sql-plan-management.md [hunk 10]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
...
+- Currently, this feature only supports binding historical execution plans in the `statements_summary` and `statements_summary_history` tables of the current TiDB node. If you get a `can't find any plans` error, you can connect to another TiDB node in the cluster and retry the binding.
+- Currently, this feature does not work for queries with subqueries, queries that access TiFlash, or queries that join 3 or more tables.
+
+The SQL statement of this binding method is as follows:
+
+```sql
+CREATE [GLOBAL | SESSION] BINDING FROM HISTORY USING PLAN DIGEST 'plan_digest';
+```
+
+This statement binds a execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. For the applicable SQL statements, priorities, scopes, and effective conditions of the created bindings, they are the same as that of [bindings created using SQL hints](#create-a-binding-according-to-a-sql-statement).
```
Comment (qiancai) [inline sql-plan-management.md:189]:
```suggestion
This statement binds a execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. For the applicable SQL statements, priorities, scopes, and effective conditions of the created bindings, they are the same as that of [bindings created using SQL statements](#create-a-binding-according-to-a-sql-statement).
```

File: sql-plan-management.md [hunk 11]
```diff
@@ -222,7 +330,9 @@ This statement outputs the execution plan bindings at the GLOBAL or SESSION leve
 | update_time | Updating time |
 | charset | Character set |
 | collation | Ordering rule |
-| source | The way in which a binding is created, including `manual` (created by the `create [global] binding` SQL statement), `capture` (captured automatically by TiDB), and `evolve` (evolved automatically by TiDB) |
+| source | The way in which a binding is created, including `manual` (created according to a SQL hint), `history` (created according to a historical execution plan), `capture` (captured automatically by TiDB), and `evolve` (evolved automatically by TiDB) |
```
Comment (qiancai) [inline sql-plan-management.md:333]:
```suggestion
| source | The way in which a binding is created, including `manual` (created according to a SQL statement), `history` (created according to a historical execution plan), `capture` (captured automatically by TiDB), and `evolve` (evolved automatically by TiDB) |
```

File: sql-plan-management.md [hunk 12]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
 >
 > For `PREPARE` / `EXECUTE` statements and for queries executed with binary protocols, you need to create execution plan bindings for the real query statements, not for the `PREPARE` / `EXECUTE` statements.
 
-### Remove binding
+#### Create a binding according to a historical execution plan
+
+To fix a SQL statement to use a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than binding it according to a SQL statement.
```
Comment (ran-huang) [inline sql-plan-management.md:171]:
```suggestion
To make the execution plan of a SQL statement fixed to a historical execution plan, you can use `plan_digest` to bind that historical execution plan to the SQL statement, which is more convenient than binding it according to a SQL statement.
```

File: sql-plan-management.md [hunk 13]
```diff
@@ -222,7 +330,9 @@ This statement outputs the execution plan bindings at the GLOBAL or SESSION leve
 | update_time | Updating time |
 | charset | Character set |
 | collation | Ordering rule |
-| source | The way in which a binding is created, including `manual` (created by the `create [global] binding` SQL statement), `capture` (captured automatically by TiDB), and `evolve` (evolved automatically by TiDB) |
+| source | The way in which a binding is created, including `manual` (created according to a SQL statement), `history` (created according to a historical execution plan), `capture` (captured automatically by TiDB), and `evolve` (evolved automatically by TiDB) |
+| sql_digest | Digest of normalized SQL statement |
```
Comment (ran-huang) [inline sql-plan-management.md:334]:
```suggestion
| sql_digest | Digest of a normalized SQL statement |
```

File: sql-plan-management.md [hunk 14]
```diff
@@ -222,7 +330,9 @@ This statement outputs the execution plan bindings at the GLOBAL or SESSION leve
...
+| plan_digest | Digest of execution plan |
```
Comment (ran-huang) [inline sql-plan-management.md:335]:
```suggestion
| plan_digest | Digest of an execution plan |
```

**Comment by qiancai** [comment]
@ran-huang All your suggested changes are accepted and committed. PTAL again. Thanks.

File: sql-plan-management.md [hunk 15]
```diff
@@ -162,7 +166,101 @@ The original SQL statement and the bound statement must have the same text after
...
+- Currently, this feature only supports binding historical execution plans in the `statements_summary` and `statements_summary_history` tables of the current TiDB node. If you get a `can't find any plans` error, you can connect to another TiDB node in the cluster and retry the binding.
+- Currently, this feature does not work for queries with subqueries, queries that access TiFlash, or queries that join 3 or more tables.
+
+The SQL statement of this binding method is as follows:
+
+```sql
+CREATE [GLOBAL | SESSION] BINDING FROM HISTORY USING PLAN DIGEST 'plan_digest';
+```
+
+This statement binds an execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. For the applicable SQL statements, priorities, scopes, and effective conditions of the created bindings, they are the same as that of [bindings created according to SQL statements](#create-a-binding-according-to-a-sql-statement).
```
Comment (ran-huang) [inline sql-plan-management.md:189]:
```suggestion
This statement binds an execution plan to a SQL statement using `plan_digest`. The default scope is SESSION. The applicable SQL statements, priorities, scopes, and effective conditions of the created bindings are the same as that of [bindings created according to SQL statements](#create-a-binding-according-to-a-sql-statement).
```

**Review by ran-huang — COMMENTED** [review]
Rest LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**